hazelcast/src/main/java/com/hazelcast/util/FutureUtil.java

```diff
@@ -246,6 +264,19 @@ public static void waitWithDeadline(Collection<Future> futures, long timeout, Ti
     }
 
     @PrivateApi
+    public static void waitUntilAllRespondedWithDeadline(Collection<Future> futures, long timeout, TimeUnit timeUnit,
+                                                         ExceptionHandler exceptionHandler) {
+        CollectAllExceptionHandler collector = new CollectAllExceptionHandler(futures.size());
+        waitWithDeadline(futures, timeout, timeUnit, collector);
+        final List<Throwable> throwables = collector.getThrowables();
+        synchronized (throwables) {
```

Comment:
Ok. Learned something new. Apparently it is part of the synchronizedList to give a guarantee on the underlying lock being used. Ans synchronized(list) is a recommended approach.